worker/cmd/generate.go

```diff
@@ -149,21 +149,19 @@ var generateCmd = &cobra.Command{
 		svc := s3.NewFromConfig(cfg)
 
 		sigChan := make(chan os.Signal, 1)
-		jobChan := make(chan string)
 		stopChan := make(chan struct{})
 
 		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
 
 		var wg sync.WaitGroup
 		wg.Add(1)
-		go func(jobChan chan<- string, stopChan <-chan struct{}) {
+		go func(stopChan <-chan struct{}) {
 			defer wg.Done()
 			timer := time.NewTicker(1 * time.Second)
 			for {
 				select {
 				case <-stopChan:
 					sugar.Info("Shutting down job listener")
-					close(jobChan)
 					return
```

Comment:
This change breaks allowing the signal handler (`sigChat`) from triggering shutdown of the worker.

Reply:
The channel that gets notified for a signal is `sigChan`. The changes here (not exact spot where the comment is) mean that while a job is processed, we are never doing a `select` on `sigChan` until the job is complete. Prior to these changes, we'd be watching that channel while the job was processing.

Reply:
Got it. Before these changes, although the first go routine will exit on os.Signal after closing the `jobChan`, but if the job is still processing, the go routing that's iterating over jobChan will stop iterating, but it won't exit, because the job is still in progress and wg.Done is called. I tested with the sleep (simulated work, not perfect, because it doesn't react to context state) and the end result is same in both the scenario, worker only exit once running process gets over.